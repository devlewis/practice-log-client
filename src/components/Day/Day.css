.day_box {
  padding-left: 5px;
  padding-right: 5px;
  border: solid black 1px;
  border-radius: 2%;
  width: 80px;
  height: 140px;
  margin-bottom: 20px;
}

.day_box .hours {
  margin-bottom: 8px;
}

.bold {
  font-weight: bold;
}

.day_box .big {
  font-size: 14px;
  margin-bottom: 0;
  margin-top: 3px;
}

.day_box p {
  font-size: 13px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
}

.small {
  font-size: 14px;
}

.dypbutton {
  margin: auto;
}

.completed_box {
  padding-right: 5px;
}

.qubox {
  margin-top: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.fa-pencil-alt {
  margin-top: 6px;
  color: white;
}

.fa-pencil-alt:hover {
  color: black;
}

.topline {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

@media screen and (min-width: 640px) {
  .day_box {
    width: 100px;
    height: 150px;
  }
}

@media screen and (min-width: 768px) {
  .day_box {
    width: 90px;
    height: 135px;
  }
}
