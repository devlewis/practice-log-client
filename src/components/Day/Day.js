import React, { PureComponent } from "react";
import { Link } from "react-router-dom";
import "../Day/Day.css";

class Day extends PureComponent {
  state = {
    hover: false,
  };

  toggleHover = () => {
    this.setState({ hover: !this.state.hover });
  };

  render() {
    let mystyle = {};

    this.state.hover
      ? (mystyle = { color: "black", backgroundColor: "aliceblue" })
      : this.props.touched && this.props.completed === "false"
      ? (mystyle = {
          color: "white",
          backgroundColor: "red",
        })
      : this.props.touched &&
        this.props.completed === "true" &&
        this.props.actual_hours < this.props.hours_goal
      ? (mystyle = {
          color: "black",
          backgroundColor: "yellow",
        })
      : this.props.touched && this.props.completed === "true"
      ? (mystyle = {
          color: "white",
          backgroundColor: "green",
        })
      : (mystyle = {});

    const day_num = this.props.day_num;
    const date = this.props.day_date;

    return (
      <Link to={`/day/${day_num}`}>
        <div className="day_box" style={mystyle}>
          <div className="bold">
            <div className="topline">
              <p className="big">Day #{day_num}</p>
              <i
                className="fas fa-pencil-alt fa-sm"
                style={mystyle}
                onMouseEnter={this.toggleHover}
                onMouseLeave={this.toggleHover}
              ></i>
            </div>
            <p className="big">{date}</p>
          </div>
          {this.props.touched === false ? (
            <div className="qubox">
              <i
                className="fas fa-question fa-3x"
                onMouseEnter={this.toggleHover}
                onMouseLeave={this.toggleHover}
              ></i>
            </div>
          ) : (
            <section className="completed_box">
              <p className="hours">
                {this.props.actual_hours}/{this.props.goal_hours} hours
              </p>
              {(this.props.technique || this.props.repertoire) && (
                <p className="bold">Details:</p>
              )}
              {this.props.technique && <p>Technique: {this.props.technique}</p>}
              {this.props.repertoire && (
                <p> Repertoire: {this.props.repertoire}</p>
              )}
            </section>
          )}
        </div>
      </Link>
    );
  }
}

export default Day;
