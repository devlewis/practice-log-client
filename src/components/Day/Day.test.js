import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Day from "./Day";

describe(`Day component`, () => {
  const props1 = {
    id: 1,
    day_num: 1,
    date: "March 22",
    completed: "false",
    technique: "",
    repertoire: "",
    actual_hours: 0,
    touched: false,
    goal_id: 1,
    user_id: 1,
    goal_hours: 3,
  };

  const props2 = {
    id: 2,
    day_num: 2,
    date: "March 23",
    completed: "true",
    technique: "Hanon",
    repertoire: "",
    actual_hours: 2,
    touched: true,
    goal_id: 1,
    user_id: 1,
    goal_hours: 3,
  };

  it("renders a .Day by default", () => {
    const wrapper = shallow(<Day />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Day given props of untouched", () => {
    const wrapper = shallow(<Day {...props1} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Day given props of touched", () => {
    const wrapper = shallow(<Day {...props2} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
