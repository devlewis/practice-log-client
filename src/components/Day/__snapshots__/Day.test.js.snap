// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component renders a .Day by default 1`] = `
<div
  className="day_box"
  style={Object {}}
>
  <p>
    Day #
  </p>
  <p />
  <section
    className="completed_box"
  >
    <p>
      Hours Practiced: 
    </p>
    <p>
      Goal Hours: 
    </p>
    <Link
      className="small"
      to="/day/undefined"
    >
      Edit Practice Details
    </Link>
  </section>
</div>
`;

exports[`Day component renders the Day given props of touched 1`] = `
<div
  className="day_box"
  style={
    Object {
      "backgroundColor": "green",
      "color": "white",
    }
  }
>
  <p>
    Day #
    2
  </p>
  <p />
  <section
    className="completed_box"
  >
    <p>
      Hours Practiced: 
      2
    </p>
    <p>
      Goal Hours: 
      3
    </p>
    <p>
      Practice Details:
    </p>
    <p>
      Technique: 
      Hanon
    </p>
    <Link
      className="small"
      to="/day/2"
    >
      Edit Practice Details
    </Link>
  </section>
</div>
`;

exports[`Day component renders the Day given props of untouched 1`] = `
<div
  className="day_box"
  style={Object {}}
>
  <p>
    Day #
    1
  </p>
  <p />
  <Link
    to="/day/1"
  >
    Did you practice today?
  </Link>
</div>
`;
